File: menu.py
import requests
from bs4 import BeautifulSoup

def fetch_menu():
    url = 'https://cgi.ravintolapalvelut.iss.fi/ravintola'
    response = requests.get(url)
    soup = BeautifulSoup(response.text, 'html.parser')

    menu_items = []

    for day_section in soup.select('.lunch-menu__day'):
        day = day_section.select_one('h2.collapsible-title')
        paragraphs = day_section.select('p')

        day_name = day.get_text(strip=True) if day else 'Unknown day'
        items = [p.get_text(strip=True) for p in paragraphs if p.get_text(strip=True)]

        menu_items.append({
            'day': day_name,
            'items': items
        })

    print(menu_items)

fetch_menu()

File: menu.js
const fs = require('fs');
const cheerio = require('cheerio');

async function fetchMenu() {
  try {
    const url = 'https://cgi.ravintolapalvelut.iss.fi/ravintola';
    const response = await fetch(url);
    const html = await response.text();
    const $ = cheerio.load(html);

    const menuItems = [];

    $('.lunch-menu__day').each((i, elem) => {
      const day = $(elem).find('h2.collapsible-title').text().trim();
      const items = [];

      $(elem).find('p').each((j, p) => {
        const text = $(p).text().trim();
        if (text) {
          items.push(text);
        }
      });

      menuItems.push({ day, items });
    });

    fs.writeFileSync('menu.json', JSON.stringify(menuItems, null, 2), 'utf8');
    console.log('Menu saved to menu.json');
  } catch (error) {
    console.error('Error fetching menu:', error);
  }
}

fetchMenu();